src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryAvroConverter.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.gcp.bigquery.sink;
+
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.plugin.common.RecordConverter;
+import io.cdap.plugin.format.avro.StructuredToAvroTransformer;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.mapred.AvroKey;
+
+import java.io.IOException;
+
+/**
+ * BigQueryAvroConverter converts a {@link StructuredRecord} to {@link AvroKey<GenericRecord>}
+ */
+public class BigQueryAvroConverter extends RecordConverter<StructuredRecord, AvroKey<GenericRecord>> {
+
+  private final StructuredToAvroTransformer structuredToAvroTransformer;
+
+  BigQueryAvroConverter() {
+    structuredToAvroTransformer = new StructuredToAvroTransformer(null);
+  }
+
+  @Override
+  public AvroKey<GenericRecord> transform(StructuredRecord record, Schema schema) throws IOException {
+    return new AvroKey<>(structuredToAvroTransformer.transform(record, schema));
```

Comment:
I would label the `schema` attribute as Nullable since Avro supports a nullable schema now.

Reply:
Done.